assignments/tasks.go

```diff
@@ -0,0 +1,284 @@
+package assignments
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"path/filepath"
+
+	pb "github.com/autograde/quickfeed/ag"
+	"github.com/autograde/quickfeed/database"
+	"github.com/autograde/quickfeed/scm"
+)
+
+// Things to do:
+// - Repositories in database do not have a "Name"-field.
+// - Ordering of tasks (See teacher.md)
+
+// newTask returns a task from markdown contents and associates it with the given assignment.
+// The provided markdown contents must contain a title specified on the first line,
+// starting with the "# " character sequence, followed by two new line characters.
+func newTask(contents []byte, assignment *pb.Assignment, name string) (*pb.Task, error) {
+	if !bytes.HasPrefix(contents, []byte("# ")) {
+		return nil, fmt.Errorf("task for assignment %s does not start with a # title marker", assignment.Name)
+	}
+	bodyIndex := bytes.Index(contents, []byte("\n\n"))
+	if bodyIndex == -1 {
+		return nil, fmt.Errorf("failed to find task body in %s", assignment.Name)
+	}
+
+	task := &pb.Task{
+		AssignmentOrder: assignment.GetOrder(),
+		Title:           string(contents[2:bodyIndex]),
+		Body:            string(contents[bodyIndex+2:]),
+		Name:            name,
+	}
+
+	return task, nil
+}
+
+// Updates an issue on specified repository
+func updateScmIssue(c context.Context, sc scm.SCM, course *pb.Course, repo *pb.Repository, issue *pb.Issue) (*scm.Issue, error) {
+	issueOptions := &scm.CreateIssueOptions{
+		Organization: course.Name,
+		Repository:   filepath.Base(repo.GetHTMLURL()), // Todo
```

Comment:
I've added a helper method to `pb.Repository` that you can use here: `repo.Name()`